services/game_service.rb

```diff
@@ -0,0 +1,108 @@
+require 'faraday'
+
+class GameService
+
+GAME_API = '177f68493b7f48e7c3deaf6d5a815ff0'
+
+  def get_games_by_name(game)
+    response = conn.get('/games') do |res|
+      res.body = "fields *; where name = \"#{game}\";"
+    end
+    body = JSON.parse(response.body, symbolize_names: true)
+  end
+  
+  def get_games_by_id(game_id)
+    response = conn.get('/games') do |res|
+      res.body = "fields *; where id = #{game_id};"
+    end
+    body = JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_games_by_keywords(ids)
+    id_list =  ids.join(",")
+    response = conn.get('/games') do |res|
+      res.body = "fields name, summary, rating, keywords; 
+                  where keywords = [#{id_list}]
+                  & rating > 1; 
+                  sort rating desc;"
+      # NOTE: [] is AND, () is OR
+    end
+    body = JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_keyid(word)
+    response = conn.get("/keywords") do |res|
+      res.body = "fields id, name; where name = \"#{word}\";" 
+    end
+    body = JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_game_videos(video_id)
```

Comment:
these methods are all VERY similar, could be refactored to DRY it up

Reply:
yeah, i was wondering if we could do like a get_url method and then pass in the url path we want